dadapy/metric_comparisons.py

```diff
@@ -611,3 +611,64 @@ def return_label_overlap_selected_coords(self, labels, coord_list, k=30):
             overlaps.append(overlap)
 
         return overlaps
+
+    def return_inf_imb_causality(
+        self, cause_present, effect_present, effect_future, weights=[1], k=1
+    ):
+        """Return the imbalances (weight * cause_present, effect_present) -> effect_future
+
+        Args:
+            cause_present (np.ndarray(float)): N x D1 matrix, putative driver system data set at time t
+            effect_present (np.ndarray(float)): N x D2 matrix, putative driven system data set at time t
+            effect_future (np.ndarray(float)): N x D2 matrix, putative driven system data set at time t + tau
+            weights (list(float), np.ndarray(float)): scaling parameters for the driver system at time t
+            k (int): order of nearest neighbour considered for the calculation of the imbalance
+
+        Returns:
+            imbalances (np.ndarray(float)): the information imbalances for the different weights
+        """
+
+        _, ranks_effect_future = compute_nn_distances(
+            effect_future, self.maxk, self.metric, self.period
+        )
+
+        imbalances = Parallel(n_jobs=self.njobs)(
+            delayed(self.return_inf_imb_causality_target_rank)(
+                cause_present, effect_present, ranks_effect_future, weight, k
+            )
+            for weight in weights
+        )
+
+        return imbalances
+
+    def return_inf_imb_causality_target_rank(
+        self, cause_present, effect_present, ranks_effect_future, weight=1, k=1
+    ):
+        """Return the imbalance (weight * cause_present, effect_present) -> effect_future
```

Comment:
Same thing here:

Return the imbalance (weight * cause_present, effect_present) -> effect_future.